lib/mongo/server/connection_pool/generation_manager.rb

```diff
@@ -45,6 +45,19 @@ def generation(service_id: nil)
           end
         end
 
+        def generation_unlocked(service_id: nil)
```

Comment:
Add `should_lock` argument to `generation` method instead of creating a whole new method. Or extract the duplicating code into a separate method.

Reply:
done